pkg/containers/containers.go

```diff
@@ -169,58 +169,59 @@ func (c *Containers) cgroupLookupUpdate(cgroupId uint64) error {
 // add cgroupId with a matching container id, extracted from path
 func (c *Containers) CgroupUpdate(cgroupId uint64, path string) (CgroupInfo, error) {
 	info := CgroupInfo{Path: path}
+	info.ContainerId, info.Runtime = getContainerIdFromCgroup(path)
 
-	for _, pc := range strings.Split(path, "/") {
-		if len(pc) < 64 {
-			continue
-		}
-		containerId, runtime := getContainerIdFromCgroup(pc)
-		if containerId == "" {
-			continue
-		}
-		info.ContainerId = containerId
-		info.Runtime = runtime
-		// we want to get container info with respect to host
-		break
-	}
 	c.mtx.Lock()
 	c.cgroups[uint32(cgroupId)] = info
 	c.mtx.Unlock()
 	return info, nil
 }
 
 // extract container id and container runtime from path
-func getContainerIdFromCgroup(pathComponent string) (string, string) {
-	runtime := "unknown"
-	path := strings.TrimSuffix(pathComponent, ".scope")
-
-	if strings.HasPrefix(path, "docker-") {
-		runtime = "docker"
-		path = strings.TrimPrefix(path, "docker-")
-		goto check
-	}
-	if strings.HasPrefix(path, "crio-") {
-		runtime = "crio"
-		path = strings.TrimPrefix(path, "crio-")
-		goto check
-	}
-	if strings.HasPrefix(path, "cri-containerd-") {
-		runtime = "containerd"
-		path = strings.TrimPrefix(path, "cri-containerd-")
-		goto check
-	}
-	if strings.HasPrefix(path, "libpod-") {
-		runtime = "podman"
-		path = strings.TrimPrefix(path, "libpod-")
-		goto check
-	}
+func getContainerIdFromCgroup(cgroupPath string) (string, string) {
+	prevPathComp := ""
+	for _, pc := range strings.Split(cgroupPath, "/") {
+		if len(pc) < 64 {
+			prevPathComp = pc
+			continue
+		}
+
+		runtime := "unknown"
+		id := strings.TrimSuffix(pc, ".scope")
+
+		if strings.HasPrefix(id, "docker-") {
```

Comment:
Maybe here we could `switch` on `strings.HasPrefix` without having to `goto check` at every case.
In effect, the break statement that is needed at the end of each case is provided automatically in Go. WDYT?